### ISSUE TITLE
Need the ability to know on which branch a tag was pushed

### ISSUE THREAD

**Issue by fleroux514** [google-github-actions/auth#46]
### TL;DR

Currenty the `ref` token attribute seems to be used to indicate on which branch a change was made in the case of a commit (ex. "ref": "refs/heads/develop"), and to get the name of the tag when a tag is pushed (ex. "ref": "refs/tags/my-tag"). 

So in the case of a tag, it's currently not possible to know on which branch it was pushed. On the Workload Identity Pool Provider side, we would like to be able to validate that a tag was pushed on the a repository's master branch in order to grant auth access on the received token.

Or else anybody could create a feature branch modifying a Github Action CI/CD deployment for example, and pushing a tag on their feature branch would allow them to actually execute the modified pipeline.

This lack of functionality is currently a blocker for us to be able to grant token access on newly created tags.

**Comment by sethvargo** [comment]
Hi @fleroux514 - thank you for opening an issue.

> So in the case of a tag, it's currently not possible to know on which branch it was pushed.

A **tag** is a pointer to a **commit**, and **commits** exist independently of **branches**. It is important to understand that tags have no direct relationship with branches - they only ever identify a commit.

> On the Workload Identity Pool Provider side, we would like to be able to validate that a tag was pushed on the a repository's master branch in order to grant auth access on the received token.

I'm sorry, but this doesn't make sense. Tags are against a commit, not a branch. The only think you could possibly do to achieve this is to extract the commit from the tag and then walk the tree for your master branch and ensure that commit hash exists in the tree.

> Or else anybody could create a feature branch modifying a Github Action CI/CD deployment for example, and pushing a tag on their feature branch would allow them to actually execute the modified pipeline.

It's a [long-standing issue](https://github.com/isaacs/github/issues/1091) and [thread](https://github.community/t/feature-request-protected-tags/1742/67) that GitHub lacks controls and policies around tags. It's worth noting that users don't need a feature branch to commit and push a tag either. My larger concern here is that you appear to have untrusted users with access to a repository and are attempt to use tags as an authorization principal (of which tags are not a good solution).

In any event, I don't think there's anything we could do in this repository to assist with your use case. We can only assert against the [claims that GItHub provides](https://token.actions.githubusercontent.com/.well-known/openid-configuration), so it's probably best to raise this request with GitHub. Sorry!
